src/heliaRemotePinner.ts

```diff
@@ -31,34 +31,72 @@ export interface HeliaRemotePinnerConfig {
    * @default { retries: 10 }
    */
   retryOptions?: pRetryOptions
+
+  /**
+   * Whether to merge the origins from the libp2p node with the provided origins.
+   * If false, it will only use the provided origins.
+   * If false and no origins are provided, it will use the libp2p node's multiaddrs.
+   * If true and no origins are provided, it will use the libp2p node's multiaddrs.
+   * If true and origins are provided, it will merge the libp2p node's multiaddrs and the provided origins.
+   *
+   * @default false
+   */
+  mergeOrigins?: boolean
```

Comment:
because in the case that `additionalOrigins.length >0`, we may want to *only* use those ("im pinning content I know I don't have locally" or "im pinning content i have locally and you can only access me at these multiaddrs").

Not just merge ("use my helia multiaddrs, but also these others")

There are a few cases we want to cover:

1. allow user to restrict multiaddrs to explicit list (providedOrigins + merge=false)
2. allow user to append multiaddrs to helia.libp2p.multiaddrs (providedOrigins + merge=true)
3. allow user to use helia.libp2p.multiaddrs only.

Option 1 & 2 require the addition of a flag instead of the implicit expectation based on the providedOrigins.